src/renderer/html_handlebars/helpers/navigation.rs

```diff
@@ -169,8 +168,67 @@ mod tests {
       h.register_helper("previous", Box::new(previous));
       h.register_helper("next", Box::new(next));
 
-      assert_eq!(h.template_render(template, &data).unwrap(), "
-         one: one.html
-         three: three.html");
+      assert_eq!(
+         h.template_render(TEMPLATE, &data).unwrap(),
+         "one: one.html|three: three.html");
+   }
+
+   #[test]
+   fn test_first() {
+      let data = json!({
+         "name": "one",
+         "path": "one.path",
+         "chapters": [
+            {
+               "name": "one",
+               "path": "one.path"
+            },
+            {
+               "name": "two",
+               "path": "two.path",
+            },
+            {
+               "name": "three",
+               "path": "three.path"
+            }
+         ]
+      });
+
+      let mut h = Handlebars::new();
+      h.register_helper("previous", Box::new(previous));
+      h.register_helper("next", Box::new(next));
+
+      assert_eq!(
```

Comment:
I usually break this out into a couple lines. So at the top of the test I'll assign what I expect to a `should_be` variable, then assign what we actually get to `got`. Then at the bottom you can write `assert_eq!(got, should_be);` (e.g. [config.rs]).

It's definitely a personal taste thing, but it's nice when you can look at the top and figure out what the expected output should be at a glance without needing to look for all `assert_eq!()` calls. 

[config.rs]: https://github.com/rust-lang-nursery/mdBook/blob/fb99276f528307238dc5c1c66f17e6524efcf4ad/src/config.rs#L296-L327